qa/vagrant/src/test/resources/packaging/utils/plugins.bash

```diff
@@ -176,6 +169,22 @@ install_and_check_plugin() {
     done
 }
 
+# Install a meta plugin
+# $1 - the plugin name
+# $@ - all remaining arguments are jars that must exist in the plugin's
+#      installation directory
+install_meta_plugin() {
+    local name=$1
+
+    install_plugin $name "$(readlink -m $name-*.zip)"
+    assert_module_or_plugin_directory "$ESPLUGINS/$name"
+
+    local user="$ESPLUGIN_COMMAND_USER"
+    local group="$ESPLUGIN_COMMAND_USER"
+
+    assert_file_exist "$ESPLUGINS/$name/meta-plugin-descriptor.properties" f $user $group 775
```

Comment:
I don't understand why we need such checks. The installer will not succeed if the plugin zip is malformed (ie missing descriptor file) so why make these tests susceptible to breaks when we change the internal format? For example, I have considered making plugins put jar files under a lib dir, which would again break these tests, but for no good reason.

Reply:
I'm not sure we need this one either. I get wanting to have some assertion early after installing the plugin but I think it is enough to make sure that the directory exists.

My take on the reason we assert that jars and stuff are actually created:
1. We assert the owner and permissions on the JAR in the real-est way possible. If any of that is important then the assertion is important.
2. We want to do some limited sanity check that the plugin is installed correctly. We could execute some API call against the plugin to prove that it is loaded properly but this is brittle in that it relies on the plugin's behavior. And it requires a running Elasticsearch. What we have here is fast but brittle for all the reasons Ryan mentioned.

I don't know the right solution. I think not asserting anything at all after installing the plugin isn't great. Even installing the plugin, starting Elasticsearch, and listing the plugins isn't really enough to ensure the plugin isn't borked. Asserting the jars was a simple thing we could do at the time but a thing I really would like to rethink.

Reply:
I used to add tests for files and directories that are accessed by Elasticsearch at runtime. This is not the case of the `meta-plugin-descriptor.properties` so I agree it can be removed.

I still think that we need some basic assertions after installing a plugin (meta or not) in the packaging tests. The most important to me are checking the permissions and owerships of plugin's bin, config, descriptor and security policy files on all linux-supported platforms. We had enough issues with file permissions/owerships on various operating systems and I don't think we're confident enough to remove these checks. I also think they are not very costly to maintain compared to the safety belt they provide.

For the plugin's jars and internal directory structure I agree it makes less sense to test them in detail. We should have better integration tests for each plugin instead to ensure the plugin isn't borked.

Reply:
> The most important to me are checking the permissions and owerships of plugin's bin, config, descriptor and security policy files on all linux-supported platforms

But we already have tests for this. By having these checks here, we are re-testing the elasticsearch-plugin install command behavior for every official plugin. That duplication is what makes this a pain to maintain.

Note this PR is fine on its own, I only wanted to suggest simplify this testing since it was slightly touched here.

Reply:
> Note this PR is fine on its own, I only wanted to suggest simplify this testing since it was slightly touched here.

Good, let's move forward then and we could revisit the duplication in a potential follow up pull request.